### PR TITLE
feat: add entry point to feed nav

### PR DESCRIPTION
**Pull request by capJavert**
## Changes

- add modal handler for plus entry point in feed nav, see [thread](https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1733310994510699)

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://mobile-new-feed-entry-point.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/tabs/TabContainer.tsx [hunk 1]
```diff
@@ -84,22 +87,25 @@ export function TabContainer<T extends string = string>({
...
-
-    setTimeout(() => {
-      if (shouldFocusTabOnChange && containerRef?.current) {
-        const [firstChild] = containerRef.current.children;
-        if (firstChild instanceof HTMLElement) {
-          firstChild.focus();
+    const shouldChange = onActiveChange?.(label, event);
+
+    // evaluate !== false due to backwards compatibility with implementations that return undefined
+    if (shouldChange !== false) {
```
Comment (capJavert) [inline packages/shared/src/components/tabs/TabContainer.tsx:96]: Added support where we can return `false` from `onActiveChange` which will cancel any tab navigation (like `event.preventDefault` for native clicks.

File: packages/shared/src/components/feeds/FeedNav.tsx [hunk 2]
```diff
@@ -154,6 +163,17 @@ function FeedNav(): ReactElement {
 
             return null;
           }}
+          onActiveChange={(label, event) => {
+            if (showPlusSubscription && label === FeedNavTab.NewFeed) {
+              event.preventDefault();
+
+              openModal({ type: LazyModal.AdvancedCustomFeedSoon, props: {} });
+
+              return false;
```
Comment (capJavert) [inline packages/shared/src/components/feeds/FeedNav.tsx:172]: Do not navigate to new feed and instead show plus modal

File: packages/shared/src/components/tabs/TabList.tsx [hunk 3]
```diff
@@ -24,7 +24,10 @@ interface DimensionProps {
 export interface TabListProps<T extends string = string> {
   items: Pick<TabProps<T>, 'label' | 'url'>[];
   active: T;
-  onClick?: (label: T) => unknown;
+  onClick?: (
+    label: T,
+    event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>,
+  ) => unknown;
```
Comment (capJavert) [inline packages/shared/src/components/tabs/TabList.tsx:30]: Exposed `event` as second param as well

**Review by rebelchris — APPROVED** [review]
Looks good to me

**Pull request merged**